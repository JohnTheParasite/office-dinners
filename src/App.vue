<template>
  <div id="app">
    <toaster-container />
    <simple-dialog />
    <menu-container v-if="showMenu"></menu-container>
    <div class="content-container">
      <div class="top-shadow"></div>
      <top-bar v-if="showMenu"></top-bar>
      <router-view />
    </div>
  </div>
</template>
<script>
import MenuContainer from "@/components/menu/MenuContainer"
import TopBar from "@/components/menu/TopBar"
import ToasterContainer from "@/components/controls/ToastContainer"
import SimpleDialog from "@/components/modals/SimpleDialog"

export default {
  components: { SimpleDialog, ToasterContainer, TopBar, MenuContainer },
  computed: {
    showMenu() {
      return this.$route.name !== "Login" && this.$route.name !== null
    }
  }
}
</script>

<style lang="scss">
@import "scss/core.scss";

#app {
  display: flex;
}

.content-container {
  width: 100%;
}

.top-shadow {
  background: linear-gradient(180deg, rgba(248, 248, 248, 0.95) 44%, rgba(248, 248, 248, 0.46) 73%, rgba(255, 255, 255, 0));
  left: 0;
  display: block;
  width: 100%;
  position: fixed;
  top: 0;
  z-index: 1;
  height: 100px;
}
</style>
