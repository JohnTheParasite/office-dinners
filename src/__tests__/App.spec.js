import { createLocalVue, shallowMount } from "@vue/test-utils"
import App from "@/App"
import VueRouter from "vue-router"

describe("Main app render", () => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)
  const router = new VueRouter()

  it("should render APP without errors", () => {
    let component = shallowMount(App, {
      localVue,
      router
    })
    expect(component).toMatchSnapshot()
  })
})
