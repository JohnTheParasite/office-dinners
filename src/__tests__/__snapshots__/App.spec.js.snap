// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main app render should render APP without errors 1`] = `
<div id="app">
  <toaster-container-stub></toaster-container-stub>
  <simple-dialog-stub></simple-dialog-stub>
  <!---->
  <div class="content-container">
    <div class="top-shadow"></div>
    <!---->
    <router-view-stub name="default"></router-view-stub>
  </div>
</div>
`;
